--- configure.old	2025-09-11 16:36:55.520706647 -0600
+++ configure.in	2025-09-11 16:42:45.361430854 -0600
@@ -1,8 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT(qmailadmin.c)
-AM_INIT_AUTOMAKE($QA_PACKAGE,$QA_VERSION, [no-define])
-AM_CONFIG_HEADER(config.h)
+AC_INIT(qmailadmin.c,[1.2.16])
+AC_CONFIG_SRCDIR(qmailadmin.c)
+AM_INIT_AUTOMAKE
+AC_CONFIG_HEADERS([config.h])
 AC_CANONICAL_HOST
 
 QA_PACKAGE="qmailadmin"
